### PR TITLE
HIVE-27180: Remove JsonSerde from hcatalog, Upgrade should update cha…

### PR DESCRIPTION
**Pull request by rtrivedi12**
…nged FQN for JsonSerDe in HMS DB

HIVE-18545 makes Hcatalog JsonSerde use the "hive.serde2" version as a back end, there are no feature differences between these implementations. This change will fix tests to use the new JsonSerDe class and remove JsonSerDe from hive-contrib. Hive Upgrade schema script should automatically update the hive table schema to rename the serde package.

### What changes were proposed in this pull request?
1. Fixed tests to use new SerDe class 'org.apache.hadoop.hive.serde2.JsonSerDe'
2. Removed JsonSerDe from hive-hcatalog.
3. Schema upgrade scripts update the SLIB column value in SERDES table to update it from "org.apache.hive.hcatalog.data.JsonSerDe" to "org.apache.hadoop.hive.serde2.JsonSerDe"

### Why are the changes needed?
Removing redundant code to make JsonSerde as first class serde in hive
Better user experience after the upgrade (Avoids CNF)

### Does this PR introduce _any_ user-facing change?
 'No'

### How was this patch tested?
mvn test -Dtest=TestMiniLlapLocalCliDriver -Dqfile=json_serde1.q,json_serde_qualified_types.q,json_serde_tsformat.q,parquet_mixed_partition_formats2.q,temp_table_parquet_mixed_partition_formats2.q

### CODE REVIEW

File: standalone-metastore/metastore-server/src/main/sql/derby/upgrade-3.2.0-to-4.0.0-alpha-1.derby.sql [hunk 1]
```diff
@@ -152,6 +152,9 @@ ALTER TABLE COMPACTION_QUEUE ADD CQ_COMMIT_TIME bigint;
 -- HIVE-24770
 UPDATE "APP".SERDES SET SLIB='org.apache.hadoop.hive.serde2.MultiDelimitSerDe' where SLIB='org.apache.hadoop.hive.contrib.serde2.MultiDelimitSerDe';
 
+-- HIVE-27180
+UPDATE "APP".SERDES SET SLIB='org.apache.hadoop.hive.serde2.JsonSerDe' where SLIB='org.apache.hive.hcatalog.data.JsonSerDe';
```
Comment (nrg4878) [inline standalone-metastore/metastore-server/src/main/sql/derby/upgrade-3.2.0-to-4.0.0-alpha-1.derby.sql:156]: we have already released 4.0.0-alpha2. So this change should go into the "4.0.0-alpha-2-to-4.0.0" upgrade script

File: standalone-metastore/metastore-server/src/main/sql/mssql/upgrade-3.2.0-to-4.0.0-alpha-1.mssql.sql [hunk 2]
```diff
@@ -176,6 +176,9 @@ ALTER TABLE COMPACTION_QUEUE ADD CQ_COMMIT_TIME bigint NULL;
 -- HIVE-24770
 UPDATE SERDES SET SLIB='org.apache.hadoop.hive.serde2.MultiDelimitSerDe' where SLIB='org.apache.hadoop.hive.contrib.serde2.MultiDelimitSerDe';
 
+-- HIVE-27180
```
Comment (nrg4878) [inline standalone-metastore/metastore-server/src/main/sql/mssql/upgrade-3.2.0-to-4.0.0-alpha-1.mssql.sql:179]: we have already released 4.0.0-alpha2. So this change should go into the "4.0.0-alpha-2-to-4.0.0" upgrade script

File: standalone-metastore/metastore-server/src/main/sql/mysql/upgrade-3.2.0-to-4.0.0-alpha-1.mysql.sql [hunk 3]
```diff
@@ -153,6 +153,9 @@ ALTER TABLE COMPACTION_QUEUE ADD CQ_COMMIT_TIME bigint;
 -- HIVE-24770
 UPDATE SERDES SET SLIB='org.apache.hadoop.hive.serde2.MultiDelimitSerDe' where SLIB='org.apache.hadoop.hive.contrib.serde2.MultiDelimitSerDe';
 
+-- HIVE-27180
```
Comment (nrg4878) [inline standalone-metastore/metastore-server/src/main/sql/mysql/upgrade-3.2.0-to-4.0.0-alpha-1.mysql.sql:156]: we have already released 4.0.0-alpha2. So this change should go into the "4.0.0-alpha-2-to-4.0.0" upgrade script

File: standalone-metastore/metastore-server/src/main/sql/oracle/upgrade-3.2.0-to-4.0.0-alpha-1.oracle.sql [hunk 4]
```diff
@@ -149,6 +149,9 @@ CREATE INDEX CTLG_NAME_DBS ON DBS(CTLG_NAME);
 -- HIVE-24770
 UPDATE SERDES SET SLIB='org.apache.hadoop.hive.serde2.MultiDelimitSerDe' where SLIB='org.apache.hadoop.hive.contrib.serde2.MultiDelimitSerDe';
 
+-- HIVE-27180
```
Comment (nrg4878) [inline standalone-metastore/metastore-server/src/main/sql/oracle/upgrade-3.2.0-to-4.0.0-alpha-1.oracle.sql:152]: we have already released 4.0.0-alpha2. So this change should go into the "4.0.0-alpha-2-to-4.0.0" upgrade script

File: standalone-metastore/metastore-server/src/main/sql/postgres/upgrade-3.2.0-to-4.0.0-alpha-1.postgres.sql [hunk 5]
```diff
@@ -277,6 +277,9 @@ ALTER TABLE "COMPACTION_QUEUE" ADD "CQ_COMMIT_TIME" bigint;
 -- HIVE-24770
 UPDATE "SERDES" SET "SLIB"='org.apache.hadoop.hive.serde2.MultiDelimitSerDe' where "SLIB"='org.apache.hadoop.hive.contrib.serde2.MultiDelimitSerDe';
 
+-- HIVE-27180
```
Comment (nrg4878) [inline standalone-metastore/metastore-server/src/main/sql/postgres/upgrade-3.2.0-to-4.0.0-alpha-1.postgres.sql:280]: we have already released 4.0.0-alpha2. So this change should go into the "4.0.0-alpha-2-to-4.0.0" upgrade script

File: hcatalog/core/src/test/java/org/apache/hive/hcatalog/data/TestJsonSerDe.java [hunk 6]
```diff
@@ -1,358 +0,0 @@
...
-import org.slf4j.LoggerFactory;
-
-
-import static org.junit.Assert.assertTrue;
-import org.junit.Test;
-
-/**
- * TestJsonSerDe.
- */
-public class TestJsonSerDe {
```
Comment (nrg4878) [inline hcatalog/core/src/test/java/org/apache/hive/hcatalog/data/TestJsonSerDe.java:50]: Shouldn't we use this test to test with the new Implementation? How would we know they are compatible with each other? We could relocate this class into a different package, but we need some tests I think.

**Review by nrg4878 — CHANGES_REQUESTED** [review]

File: standalone-metastore/metastore-server/src/main/sql/derby/upgrade-3.2.0-to-4.0.0-alpha-1.derby.sql (hunk 1 above)
Comment (rtrivedi12) [reply standalone-metastore/metastore-server/src/main/sql/derby/upgrade-3.2.0-to-4.0.0-alpha-1.derby.sql:156]: Ohh yes, that makes sense! Thanks for the review. Will update it.

File: standalone-metastore/metastore-server/src/main/sql/mssql/upgrade-3.2.0-to-4.0.0-alpha-1.mssql.sql (hunk 2 above)
Comment (rtrivedi12) [reply standalone-metastore/metastore-server/src/main/sql/mssql/upgrade-3.2.0-to-4.0.0-alpha-1.mssql.sql:179]: Will incorporate this change.

File: standalone-metastore/metastore-server/src/main/sql/mysql/upgrade-3.2.0-to-4.0.0-alpha-1.mysql.sql (hunk 3 above)
Comment (rtrivedi12) [reply standalone-metastore/metastore-server/src/main/sql/mysql/upgrade-3.2.0-to-4.0.0-alpha-1.mysql.sql:156]: Will incorporate this change.

File: standalone-metastore/metastore-server/src/main/sql/oracle/upgrade-3.2.0-to-4.0.0-alpha-1.oracle.sql (hunk 4 above)
Comment (rtrivedi12) [reply standalone-metastore/metastore-server/src/main/sql/oracle/upgrade-3.2.0-to-4.0.0-alpha-1.oracle.sql:152]: Will incorporate this change.

File: standalone-metastore/metastore-server/src/main/sql/postgres/upgrade-3.2.0-to-4.0.0-alpha-1.postgres.sql (hunk 5 above)
Comment (rtrivedi12) [reply standalone-metastore/metastore-server/src/main/sql/postgres/upgrade-3.2.0-to-4.0.0-alpha-1.postgres.sql:280]: Will incorporate this change.

File: hcatalog/core/src/test/java/org/apache/hive/hcatalog/data/TestJsonSerDe.java (hunk 6 above)
Comment (rtrivedi12) [reply hcatalog/core/src/test/java/org/apache/hive/hcatalog/data/TestJsonSerDe.java:50]: We have another extensive test suite TestJsonSerDe under org.apache.hadoop.hive.serde2 for testing the new implementation covering all test scenarios.

Comment (rtrivedi12) [reply hcatalog/core/src/test/java/org/apache/hive/hcatalog/data/TestJsonSerDe.java:50]: Added Hcatalog Test suite to new serde2 TestJsonSerDe class. Replaced Hcatalog classes with Gson classes for Json comparison.

**Pull request closed without merging**